### PR TITLE
Remove duplicated issuers in full chain

### PR DESCRIPTION
**Pull request by cipherboy**
This fixes full chain construction so that we don't accidentally introduce duplicate issuers into the chain. This also removes our incorrect setting of the legacy `IssuingCA` field, which was used prior to `CAChain` introduction.

Strictly, I believe this is caused by `CAChain` containing not just the strict issuer hierarchy but also the leaf cert. I believe this makes the most sense from a bundle perspective (at the cost of a duplication), but perhaps it is worth trimming it out.

Additionally, we haven't yet switched the old logic (for `certificate` field on the `ca_chain` endpoint) to the new full chain. I'd like to make that change, given the goals of rotation, and have it be for the default issuer only.

Thoughts?

### CODE REVIEW

**Review by stevendpclark — APPROVED** [review]

File: builtin/logical/pki/storage.go [hunk 1]
```diff
@@ -575,7 +575,6 @@ func fetchCertBundleByIssuerId(ctx context.Context, s logical.Storage, id issuer
 
 	var bundle certutil.CertBundle
 	bundle.Certificate = issuer.Certificate
-	bundle.IssuingCA = issuer.CAChain[0]
```
Comment (kitography) [inline builtin/logical/pki/storage.go:578]: sorry, this is used in sdk/helper/certutil/types.go, I think?  Clarification that  `pemBlock, _ = pem.Decode([]byte(c.IssuingCA))` will still work?

**Review by kitography — COMMENTED** [review]
Bug fix and tests look good, have question about effect on SDK

File: builtin/logical/pki/storage.go (hunk 1 above)
Comment (cipherboy) [reply builtin/logical/pki/storage.go:578]: Note its gated on size tho:

https://github.com/hashicorp/vault/blob/main/sdk/helper/certutil/types.go#L210-L212

So it won't really matter, we'll skip it.

**Comment by cipherboy** [comment]
Manually merged, per out of band conversation with @kitography and @stevendpclark. Thanks!

**Comment by cipherboy** [comment]
This PR was merged in #15277. See that PR and the relevant docs PR #15238 for more information about this change.

**Pull request closed without merging**